discovery/kubernetes/kubernetes.go

```diff
@@ -206,35 +206,29 @@ func New(l log.Logger, conf *SDConfig) (*Discovery, error) {
 		if conf.TLSConfig.CertFile != "" || conf.TLSConfig.KeyFile != "" {
 			level.Warn(l).Log("msg", "Configured TLS client certificate is ignored when using pod service account")
 		}
-		if conf.BearerToken != "" {
+		if conf.BearerToken != "" || conf.BearerTokenFile != "" {
 			level.Warn(l).Log("msg", "Configured auth token is ignored when using pod service account")
 		}
 		if conf.BasicAuth != nil {
 			level.Warn(l).Log("msg", "Configured basic authentication credentials are ignored when using pod service account")
 		}
-	} else {
-		kcfg = &rest.Config{
-			Host: conf.APIServer.String(),
-			TLSClientConfig: rest.TLSClientConfig{
-				CAFile:   conf.TLSConfig.CAFile,
-				CertFile: conf.TLSConfig.CertFile,
-				KeyFile:  conf.TLSConfig.KeyFile,
-				Insecure: conf.TLSConfig.InsecureSkipVerify,
-			},
+		if conf.ProxyURL.URL != nil {
+			level.Warn(l).Log("msg", "Configured proxy URL ignored when using pod service account")
 		}
-		token := string(conf.BearerToken)
-		if conf.BearerTokenFile != "" {
-			bf, err := ioutil.ReadFile(conf.BearerTokenFile)
-			if err != nil {
-				return nil, err
-			}
-			token = string(bf)
+	} else {
+		rt, err := config_util.NewRoundTripperFromConfig(config_util.HTTPClientConfig{
```

Comment:
Inline it like we do in https://github.com/prometheus/prometheus/blob/12708acd1516e0d1756006a54684ccaa76f1aaf2/config/config.go#L344, this would also be fragile.